docs/reference2/sql-syntax/data-types/numeric.md

```diff
@@ -30,6 +28,18 @@ The `double` data type is an inexact, variable-precision numeric type. The term
 "inexact" means an approximate value is stored. Storing values outside of its
 bounds of capacity will result in an error.
 
+## Valid ranges
+
+Numeric data types have the same valid minimum and maximum values as their
+corresponding Java types. The following table shows valid ranges for the
+integer and floating-point types. 
+
+| type     | minimum value | maximum value
+|----------|---------------|--------------
+| `int`    | [Integer.MIN_VALUE](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html#MIN_VALUE) = -2<sup>31</sup> | [Integer.MAX_VALUE](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html#MAX_VALUE) = 2<sup>31</sup>-1
+| `bigint` | -2<sup>Integer.MAX_VALUE</sup> (exclusive) | 2<sup>Integer.MAX_VALUE</sup> (exclusive)
```

Comment:
I don't believe `bigint` has a min/max range. It's only bounded by resources.

Reply:
it maps to a Java Long internally - a signed 64-bit integer. So _almost_ unlimited, has a lot of digits!

Reply:
Oh wow, today I learned. I thought we used `java.math.BigInt` under the hood. Looks like you're right:

```
ksql> insert into s1 (x) values (9223372036854775807);
ksql> select * from s1 emit changes;
+---------------------------------------------------------------------------------------------------------------------------+
|X                                                                                                                          |
+---------------------------------------------------------------------------------------------------------------------------+
|9223372036854775807                                                                                                        |
^CQuery terminated
ksql> insert into s1 (x) values (9223372036854775808);
Failed to prepare statement: line 1:29: Invalid numeric literal: 9223372036854775808
```

Thanks Nick!